functions/common/geode-common/src/main/java/org/springframework/cloud/fn/common/geode/GeodeClientCacheConfiguration.java

```diff
@@ -0,0 +1,154 @@
+				&& StringUtils.hasText(securityProperties.getPassword())) {
+
+			properties
+					.setProperty(SECURITY_CLIENT,
+							GeodeSecurityProperties.UserAuthInitialize.class.getName() + ".create");
+			properties.setProperty(SECURITY_USERNAME, securityProperties.getUsername());
+			properties.setProperty(SECURITY_PASSWORD, securityProperties.getPassword());
+
+		}
+
+		if (sslProperties.isSslEnabled()) {
+			pb.add(properties);
+			pb.add(this.toGeodeSslProperties(sslProperties));
+		}
+
+		ClientCacheFactory clientCacheFactory = new ClientCacheFactory(pb.build());
+
+		if (clientCacheProperties.isPdxReadSerialized()) {
+			clientCacheFactory.setPdxSerializer(new ReflectionBasedAutoSerializer(".*"));
+			clientCacheFactory.setPdxReadSerialized(true);
+		}
+
+		if (geodePoolProperties.getConnectType().equals(GeodePoolProperties.ConnectType.locator)) {
+			for (InetSocketAddress address : geodePoolProperties.getHostAddresses()) {
+				clientCacheFactory.addPoolLocator(address.getHostName(), address.getPort());
+			}
+		}
+		else {
+			for (InetSocketAddress address : geodePoolProperties.getHostAddresses()) {
+				clientCacheFactory.addPoolServer(address.getHostName(), address.getPort());
+			}
+		}
+
+		clientCacheFactory.setPoolSubscriptionEnabled(geodePoolProperties.isSubscriptionEnabled());
+		ClientCache clientCache = clientCacheFactory.create();
+		clientCache.readyForEvents();
+		return clientCache;
+	}
+
+	/**
+	 * Converts the App Starter properties into Geode native SSL properties.
+	 * @param sslProperties App starter properties.
+	 * @return Returns the geode native SSL properties.
+	 */
+	private Properties toGeodeSslProperties(GeodeSslProperties sslProperties) {
+
+		PropertiesBuilder pb = new PropertiesBuilder();
+
+		// locator - SSL communication with and between locators
+		// server - SSL communication between clients and servers
+		pb.setProperty("ssl-enabled-components", "server,locator");
+
+		pb.setProperty("ssl-keystore", this.resolveRemoteStore(sslProperties.getKeystoreUri(),
+				sslProperties.getUserHomeDirectory(), GeodeSslProperties.LOCAL_KEYSTORE_FILE_NAME));
+		pb.setProperty("ssl-keystore-password", sslProperties.getSslKeystorePassword());
+		pb.setProperty("ssl-keystore-type", sslProperties.getKeystoreType());
+
+		pb.setProperty("ssl-truststore", this.resolveRemoteStore(sslProperties.getTruststoreUri(),
+				sslProperties.getUserHomeDirectory(), GeodeSslProperties.LOCAL_TRUSTSTORE_FILE_NAME));
+		pb.setProperty("ssl-truststore-password", sslProperties.getSslTruststorePassword());
+		pb.setProperty("ssl-truststore-type", sslProperties.getTruststoreType());
+
+		pb.setProperty("ssl-ciphers", sslProperties.getCiphers());
+
+		return pb.build();
+	}
+
+	/**
+	 * Copy the Trust or Key stores from a remote URI into a local file.
+	 *
+	 * @param sslProperties SSL gemfire properties required for retrieving the provided
+	 *     truststore.
+	 * @return Returns the absolute path of the local trust store copy
+	 */
+	/**
```

Comment:
There is some problem with this JavaDoc, or method is missed.
No blank lines in the method JavaDocs, please.
Also I wonder what spring-boot-geode project provides for us to configure SSL...
Why do we need so much custom boilerplate code?..

Reply:
I'm using the standard checkstyle configuration. Do you want to make a change?

Reply:
This doesn't use the SDG ClientCacheFactoryBean.  We could, but there's a boat load more config properties. We would expose all of them as `spring.data.gemfire...` properties.